mongodbatlas/resource_mongodbatlas_cloud_provider_access.go

```diff
@@ -107,9 +108,14 @@ func resourceMongoDBAtlasCloudProviderAccessRead(d *schema.ResourceData, meta in
 	ids := decodeStateID(d.Id())
 	projectID := ids["project_id"]
 
-	roles, _, err := conn.CloudProviderAccess.ListRoles(context.Background(), projectID)
+	roles, resp, err := conn.CloudProviderAccess.ListRoles(context.Background(), projectID)
 
 	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+
```

Comment:
[nit] These 3 lines are used in other places as well. Maybe we can wrap this piece of code in a function a call it where is needed.

Reply:
I think this validation is so little that even If we create a single function, we will need to verify if return an error and will code exactly other 3 lines
e.g.

```
 func validateResponce(resp  *mongodbatlas.Response) err {
 if resp != nil && resp.StatusCode == http.StatusNotFound {
			d.SetId("")
			return nil
		}
 return fmt.Errorf("is not empty")
 }
 
 
 
 if err:=validateResponse(resp), err==nil{
 return nil
 }
 

```

what do you think @andreaangiolillo ?

Reply:
I want to avoid overengineering :D

Reply:
SGTM, mine was more something to think of. Thanks!